### ISSUE TITLE
systemd-shutdown disarms watchdog after completion

### ISSUE THREAD

**Issue by andreassixa** [systemd/systemd#33075]
### systemd version the issue has been seen with

254.4

### Used distribution

Embedded Linux

### Linux kernel version used

6.1.55

### CPU architectures issue was seen on

aarch64

### Component

systemd

### Expected behaviour you didn't see

The hardware watchdog should still be left active/running after completing systemd-shutdown.

### Unexpected behaviour you saw

The call to watchdog_free_device in src/shutdown/shutdown.c will disarm the watchdog with the commit in f81048f8.

watchdog_free_device calls watchdog_set_device(NULL) and that will disarm the current watchdog.

### Steps to reproduce the problem

Reboot system with hardware watchdog enabled. After systemd-shutdown completes the hardware watchdog will be disabled.

**Comment by andreassixa** [comment]
Maybe this could work? I'm not that familiarized with systemd in general.

```
int watchdog_set_device(const char *path) {
        int r;

        r = free_and_strdup(&watchdog_device, path);
        if (r > 0) { /* watchdog_device changed */
                if (!path)
                        watchdog_close(/* disarm= */ true);
                else
                        watchdog_fd = safe_close(watchdog_fd);

        return r;
}
```

**Comment by poettering** [comment]
My proposed fix is in #33079. Any chance you can give that a whirl, and test if it fixes your problem? You only need the 3rd commit.

**Comment by andreassixa** [comment]
I have tested your fix in #33079 and it works! The hardware watchdog is not disarmed anymore by systemd-shutdown. 

The test was done on systemd version v255.4 (this is the latest version I can test with my embedded Linux system at the moment).

Thanks for the help @poettering!
